lib/services/publish-service.ts

```diff
@@ -0,0 +1,208 @@
+                    } else {
+                        var chosenConnectionIndex = _.findIndex(this.allPublishConnections, c => c.name + '(' + c.publishUrl + ')' === chosenConnectionNameUrl);
+                        this.allPublishConnections.splice(chosenConnectionIndex, 1);
+                        message = util.format("Connection '%s' successfully removed.", chosenConnectionNameUrl);
+                    }
+                } else if (connectionsToBeRemoved.length === 1) {
+                    _.remove(this.allPublishConnections, connectionsToBeRemoved[0]);
+                    message = util.format("Connection '%s' successfully removed.", connectionsToBeRemoved[0].name);
+                } else {
+                    this.$errors.failWithoutHelp("No connections named '%s' found.", name);
+                }
+            }
+
+            this.savePublishJsonFile().wait();
+            this.$logger.info(message);
+        }).future<void>()();
+    }
+
+    private readConnections(): IFuture<void> {
+        return (() => {
+            var projectDir = this.$project.getProjectDir().wait();
+            var publishFilePath = path.join(projectDir, PublishService.JSON_PUBLISH_FILE_NAME);
+            try {
+                this.allPublishConnections = this.$fs.readJson(publishFilePath).wait();
+            } catch (err) {
+                if (err.code === 'ENOENT') {
+                    this.allPublishConnections = [];
+                    return;
+                }
+
+                this.$errors.failWithoutHelp("The project file %s is corrupted." + os.EOL +
+                        "Consider restoring an earlier version from your source control or backup." + os.EOL +
+                        "Additional technical info: %s", publishFilePath, err.toString());
+            }
+        }).future<void>()();
+    }
+
+    private savePublishJsonFile(): IFuture<void> {
+        return (() => { 
+            var projectDir = this.$project.getProjectDir().wait();
+            this.$fs.writeJson(path.join(projectDir, PublishService.JSON_PUBLISH_FILE_NAME), this.allPublishConnections).wait();
+        }).future<void>()()
+    }
+
+    private listConnections(connections: IPublishConnection[]): void {
+        if (!connections.length) {
+            this.$logger.info("No connections found. To add a connection, run `connection add`");
+            return;
+        }
+
+        var table = new Table({
+                head: ["Index", "Name", "Publish URL"]
+            });
+
+        _.forEach(connections, (connection, index) => {
+            table.push([index + 1, connection.name, connection.publishUrl]);
+        });
+
+        this.$logger.out(table.toString());
+    }
+
+    private validateId(id: number): void {
+        if (this.ftpPublishConnections.length <= id) {
+            this.$errors.failWithoutHelp("You only have %s connections available", this.ftpPublishConnections.length);
```

Comment:
I am guessing this is a validation message when the user sets the connection by ID? And the ID exceeds the number of stored connections? Just some random ramblings - but what happens when the user specifies an invalid ID, say 0 or a negative number? Or not an integer?

If I've figured this out correctly, you can show the following message. "Incorrect ID. For a complete list of the available connections and their IDs, run $ appbuilder publish list"